site-src/geps/gep-1282.md

```diff
@@ -0,0 +1,100 @@
+  * Openshift’s Route resource allows the [configuration of reencryption](https://docs.openshift.com/container-platform/4.10/networking/routes/secured-routes.html#nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate_secured-routes) specifically, along with custom certificate details.
+  * Contour’s HTTPProxy captures TLS details using an Envoy client certificate, destination CA certificate, and optional SubjectName which sets what Envoy should expect to see from the backend service, all inside the HTTPProxy resource. It also requires either a Protocol field inside the HTTProxy, or an annotation on the Service that tells Contour that the service expects TLS. This is [all documented](https://projectcontour.io/docs/v1.21.1/config/upstream-tls/), but I should note that Contour’s docs use the Envoy convention where a backend in Gateway parlance is called an Upstream (which may be confusing if you’re not used to it).
+  * Linkerd uses a [Server resource](https://linkerd.io/2.11/reference/authorization-policy/#server) (which is functionally pretty similar to Service in that it associates a name with a Pod selector, but also has other details like if the service supports Proxy protocol), along with a ServerAuthorization resource that specifies some constructs that sit more at the service mesh level, including identity and access control.
+
+In terms of other implementations existing use cases for features like this:
+- For Contour, there are annotations to allow the configuration of the following on a backend Service:
+  - Max connections
+  - Max pending requests
+  - Max requests
+  - Max retries
+  - Upstream protocol: This is what allows Contour to handle the 
+- In Istio, the `DestinationRule` resource allows the configuration of many settings like this:
+  - Load balancer algorithms
+  - Connection pool settings
+  - Outlier Detection
+  - Tunnel settings for non-HTTP workloads
+
+The properties we're talking about all share two things:
+- They are tightly bound to the backend, rather than being important at the route level
+- And the service owner (application developer in the Gateway API personas) should control these settings, not the owner of the Gateway implementation (the Cluster Operator)
+
+A good example is the load balancing algorithms for backends, as distinct from load balancing for routes. Load balancing for routes is about choosing which route option the implementation should choose, while load balancing for backends is about which endpoint from the backend the implementation should send traffic to if that backend's route is chosen. It's likely that these will need to be set differently (for example, random load balancing at the route level for canarying between services, with Weighted Least Request at the backend level for the actual endpoints).
```

Comment:
This distinction is pretty important, and plays in with the mesh use case quite a bit. I tried to take a stab at this previously and we decided on policy attachment, it will be very important where we go from here. No action required, just highlighting this.

Reply:
Worth cross-linking to prior discussion in https://github.com/kubernetes-sigs/gateway-api/pull/993 and https://github.com/kubernetes-sigs/gateway-api/issues/992?

Reply:
sure

Reply:
I've added in references to this discussion.